doc_source/related/dev.md

```diff
@@ -0,0 +1,409 @@
+
+```bash
+vagrant init thirtytwobits/libuavcan_v1 --box-version 1.0
+vagrant up
+```
+
+### Vagrant Cheatsheet
+
+```
+vagrant up
+vagrant ssh
+```
+
+Attach a CAN probe to the guest using VBoxManage:
+
+```
+# first, find your vagrant guest.
+VBoxManage list vms
+
+# next, find the UUID of the usb device on the host. In this example
+# we look for a PCAN probe:
+VBoxManage list usbhost | grep --context=7 PCAN
+
+# now, attach the usb device to the guest.
+VBoxManage controlvm my_libuavcan_vagrant_guest_3249032840 usbattach a2d153de-63f1-411d-08d2-eefc42b792fa
+
+# You can also setup a filter so this device will be auto-captured:
+VBoxManage usbfilter add 0 --target my_libuavcan_vagrant_guest_3249032840 --name PEAK --action hold --product "PCAN-USB Pro FD"
+```
+
+## Linux
+
+> TODO: document common SocketCAN setups for running linux examples.
+
+Libuavcan Library Developer Continuous Integration {#LibCIGuide}
+====================================================================
+
+> TODO buildkite documentation
+
+## Pi worker
+
+### Hardware
+
+1. [Rasberry PI 3 B+](https://www.newark.com/raspberry-pi/rpi3-modbp/sbc-arm-cortex-a53-1gb-sdram/dp/49AC7637) with 3A power supply.
+2. 16GB SDHC class 4 card.
+
+### Software
+
+Raspbian Buster Lite
+
+- Version: June 2019
+- Release date: 2019-06-20
+- Kernel version: 4.19
+- SHA-256: 9009409a9f969b117602d85d992d90563f181a904bc3812bdd880fc493185234
+
+### Post Install Steps
+
+#### raspi-config
+```bash
+sudo raspi-config
+```
+
+1. Change the default password. Keep this password to yourself.
+2. Enable SSH
+3. Expand the filesystem to use the entire SD card.
+4. Change the hostname to something useful and unique (this will appear as the agent name in buildkite. It must be changed from the default).
+
+#### security
+
+Some hardening steps.
+
+```bash
+cd /etc/sysctl.d
+sudo vi 98-rpi.conf
+```
+
+In 98-rpi.conf add:
+
+```bash
+fs.protected_hardlinks = 1
+fs.protected_symlinks = 1
+```
+
+then reboot.
+
+#### Software
+
+```bash
+sudo apt update
+sudo apt upgrade -y
+sudo apt install -y vim
+sudo apt install -y git
+sudo apt install -y python3-pip
+sudo apt install -y can-utils
+git clone https://github.com/thirtytwobits/nanaimo.git
+cd nanaimo
+sudo pip3 install --system .
+echo "alias la='ls -lah'" | sudo tee -a /etc/bash.bashrc
+```
+
+#### CAN
+
+Load vcan on boot:
+
+```bash
+echo vcan | sudo tee -a /etc/modules
+```
+
+Configure vcan0, and if using a can adapter, can0 on boot:
+
+```bash
+sudo vi /etc/network/interfaces.d/can
+```
+
+```bash
+auto vcan0
+	iface vcan0 inet manual
+	pre-up /sbin/ip link add dev $IFACE type vcan
+	up /sbin/ifconfig $IFACE up
+
+auto can0
+	iface can0 inet manual
+	pre-up /sbin/ip link set $IFACE type can bitrate 1000000 dbitrate 2000000 fd on sample-point .8 dsample-point .8 berr-reporting off fd-non-iso off restart-ms 100
+	up /sbin/ifconfig $IFACE up
+	down /sbin/ifconfig $IFACE down
+
+```
+
+#### [Segger JLink](https://www.segger.com/downloads/jlink)
```

Comment:
Many a drone developer is a big believer in Black Magic Probe and derivatives like Dronecode Probe so they should be at least mentioned. They are nearly feature-identical and require no daemon on the host side.

Reply:
I just don't have any experience with them so I can't write a HOWTO. I'll add a section acknowledging this.